src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -630,10 +630,12 @@ object AppDefinition extends GeneralPurposeCombinators {
       !(app.residency.isDefined ^ app.persistentVolumes.nonEmpty)
     }
 
-  private val complyWithResourceRoleRules: Validator[AppDefinition] =
-    isTrue("""Resident apps may not define acceptedResourceRoles other than "*" (unreserved resources)""") { app =>
-      def hasResidencyCompatibleRoles = app.acceptedResourceRoles.fold(true)(_ == Set(ResourceRole.Unreserved))
-      !app.isResident || hasResidencyCompatibleRoles
+  def validAcceptedResourceRoles(isResident: Boolean): Validator[Set[String]] =
```

Comment:
@aquamatthias: I've moved this validator to `ResourceRole`, and also moved the tests to a `ResourceRoleTest`. Does that look good to you?